src/planning.py

```diff
@@ -101,12 +105,35 @@ def sesame_plan(
             predicates |= new_predicates
 
 
-def task_plan(
+def task_plan_grounding(
     init_atoms: Set[GroundAtom],
     objects: Set[Object],
-    goal: Set[GroundAtom],
     strips_ops: Sequence[STRIPSOperator],
     option_specs: Sequence[OptionSpec],
+) -> List[_GroundNSRT]:
+    """Ground all operators for task planning into dummy _GroundNSRTs,
+    filtering out ones that are unreachable or have empty effects."""
+    nsrts = utils.ops_and_specs_to_dummy_nsrts(strips_ops, option_specs)
+    ground_nsrts = []
+    for nsrt in nsrts:
+        for ground_nsrt in utils.all_ground_nsrts(nsrt, objects):
+            ground_nsrts.append(ground_nsrt)
+    nonempty_ground_nsrts = [
+        nsrt for nsrt in ground_nsrts if nsrt.add_effects | nsrt.delete_effects
+    ]
+    all_reachable_atoms = utils.get_reachable_atoms(ground_nsrts, init_atoms)
```

Comment:
Here, we're using ground_nsrts as the first argument, whereas before it was nonempty_ground_nsrts. Intentional?

Reply:
it wouldn't make a difference, but not intentional, will fix